src/python/py_tokenizer.cpp

```diff
@@ -44,21 +44,40 @@ void init_tokenizer(py::module_& m) {
             return std::make_unique<ov::genai::Tokenizer>(tokenizer_path, kwargs_properties);
         }), py::arg("tokenizer_path"), py::arg("properties") = ov::AnyMap({}))
 
-        .def("encode", [](Tokenizer& tok, std::vector<std::string>& prompts, bool add_special_tokens) {
+        .def("encode", [](Tokenizer& tok, std::vector<std::string>& prompts, 
+                          bool add_special_tokens, 
+                          bool pad_to_max_length,
+                          std::optional<size_t> max_length) {
```

Comment:
should not we wrap all parameters with std::optional? to be aligned with C++ where we don't have default arguments for `encode()`

Reply:
In our approach with AnyMap/ov::Parameters indeed we have optional arguments. When user calls "encode" without arguments we come to `set_state_if_necessary` with empty anymap and optional max_length remains without value, while pad_to_max_length_val will be false.

https://github.com/openvinotoolkit/openvino.genai/blob/2f6c08c9615f212dac91f31d76880daa8d5371fa/src/cpp/src/tokenizer.cpp#L84-L90